eng/targetingpacks.targets

```diff
@@ -115,6 +115,8 @@
                        Condition="'$(UsePackageDownload)' == 'true' and $([System.String]::Copy('%(Identity)').EndsWith('Microsoft.DotNet.ILCompiler'))" />
       <PackageReference Remove="@(PackageReference)"
                         Condition="'$(UsePackageDownload)' != 'true' and $([System.String]::Copy('%(Identity)').EndsWith('Microsoft.DotNet.ILCompiler'))" />
+      <PackageReference Remove="@(PackageReference)"
+                        Condition="'%(Identity)' == 'Microsoft.DotNet.ILCompiler'" />
```

Comment:
Shouldn't this already be covered by the above four lines? If not, should we change the above four lines instead?

Reply:
The hosting pack (`runtime.win-x64.Microsoft.DotNet.ILCompiler`) and the like are covered by the above cases. However, `Microsoft.DotNet.ILCompiler` is always referenced as a PackageReference it seems. As a result, when `UsePackageDownload` is true, we need this case as well.